src/lib/Syntax.hs

```diff
@@ -91,6 +97,11 @@ data Expr = App Atom Atom
           | Atom Atom
           | Op  Op
           | Hof Hof
+          -- Add/remove fields from a record/variant, always to/from the left.
+          | RecordCons   (LabeledItems Atom) Atom   -- Add a value to a record.
+          | RecordSplit  (LabeledItems Type) Atom   -- Extract subset of fields.
+          | VariantLift  (LabeledItems Type) Atom   -- Extend a variant.
+          | VariantSplit (LabeledItems Type) Atom   -- Partition a variant.
```

Comment:
Can't those things be `PrimOp`s? I think that they would get a bunch of traversals for free if you did that.

Reply:
I vaguely remember having some reason to not do that, but I don't remember what it is. Maybe it would work fine. I can try it and see what happens.

Reply:
Looks like it works fine!